imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -50,203 +47,207 @@ def hit_l1a(packet_file: str, data_version: str) -> list[xr.Dataset]:
-            #  Will need to approach this differently
-        else:
-            raise Exception(f"Unknown APID [{apid}]")
-        metadata_arrays = defaultdict(list)
-        for packet in data_packets:
-            # Add metadata to an array
-            for field in fields(packet):
-                field_name = field.name
-                field_value = getattr(packet, field_name)
-                # convert key to lower case to match SPDF requirement
-                data_key = field_name.lower()
-                metadata_arrays[data_key].append(field_value)
-
-        # Convert integers into datetime64[s]
-        epoch_converted_times = met_to_j2000ns(metadata_arrays["shcoarse"])
-
-        # Create xarray data arrays for dependencies
-        epoch_time = xr.DataArray(
-            epoch_converted_times,
-            name="epoch",
-            dims=["epoch"],
-            attrs=attr_mgr.get_variable_attributes("epoch"),
-        )
-
-        adc_channels = xr.DataArray(
-            np.arange(64, dtype=np.uint16),
-            name="adc_channels",
-            dims=["adc_channels"],
-            attrs=attr_mgr.get_variable_attributes("adc_channels"),
-        )
-
-        # NOTE: LABL_PTR_1 should be CDF_CHAR.
-        adc_channels_label = xr.DataArray(
-            adc_channels.values.astype(str),
-            name="adc_channels_label",
-            dims=["adc_channels_label"],
-            attrs=attr_mgr.get_variable_attributes("adc_channels_label"),
-        )
-
-        # Create xarray dataset
-        dataset = xr.Dataset(
-            coords={
-                "epoch": epoch_time,
-                "adc_channels": adc_channels,
-                "adc_channels_label": adc_channels_label,
-            },
-            attrs=attr_mgr.get_global_attributes(logical_source),
+    # Drop variables not needed for CDF
+    dataset = dataset.drop_vars(drop_keys)
+
+    # Create data arrays for dependencies
+    adc_channels = xr.DataArray(
+        np.arange(64, dtype=np.uint16),
+        name="adc_channels",
+        dims=["adc_channels"],
+        attrs=attr_mgr.get_variable_attributes("adc_channels"),
+    )
+
+    # NOTE: LABL_PTR_1 should be CDF_CHAR.
+    adc_channels_label = xr.DataArray(
+        adc_channels.values.astype(str),
+        name="adc_channels_label",
+        dims=["adc_channels_label"],
+        attrs=attr_mgr.get_variable_attributes("adc_channels_label"),
+    )
+
+    # Update dataset coordinates and attributes
+    dataset = dataset.assign_coords(
+        {
+            "adc_channels": adc_channels,
+            "adc_channels_label": adc_channels_label,
+        }
+    )
+    dataset.attrs = attr_mgr.get_global_attributes(logical_source)
+
+    # Stack 64 leak variables (leak_00, leak_01, ..., leak_63)
+    dataset = concatenate_leak_variables(dataset)
+
+    # Assign attributes and dimensions to each data array in the Dataset
+    for field, data in dataset.data_vars.items():
```

Comment:
I think this will basically be re-creating all of your `DataArray`s and resetting the dataset.

Instead could you loop through just the field values and update the current dataset variable?

```python
dataset[field].attrs = attr_mgr.get_variable_attributes(field)
# Something like this where you can just set the name of existing coords
dataset[field].assign_coords(dims)
```

Reply:
Good catch. Updated!